dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfTest.scala

```diff
@@ -93,6 +93,76 @@ class BinaryOutcomeDLCWithSelfTest extends BitcoinSAsyncTest {
     }
   }
 
+  def closeTogether(num1: Double, num2: Double): Boolean = {
+    Math.abs(num1 - num2) < 1.0e-6
+  }
+
+  it should "correctly subtract fees proportionally amongst outputs" in {
+    // Can't use TransactionGenerators.realisiticOutputs as that can return List.empty
+    val nonEmptyRealisticOutputsGen = Gen
+      .choose(1, 5)
+      .flatMap(n => Gen.listOfN(n, TransactionGenerators.realisticOutput))
+      .suchThat(_.nonEmpty)
+
+    // CurrencyUnitGenerator.feeRate gives too high of fees
+    val feeRateGen = Gen.choose(0, CurrencyUnits.oneBTC.satoshis.toLong).map {
```

Comment:
I think this would probably be a useful generator to add in `CurrencyUnitGenerator`? Not sure what a good name is tho.